File: Binder/BinderUI/src/app/product/Product.ts
import {NgForm,FormGroup,FormControl,Validators,FormBuilder } from '@angular/forms'


export class Product{
    name: string="";
    description: string="";
    price: number=0;
    brand: string="";
    category:string="";
    quantity:string="";
    image:string=""
    formUserGroup: FormGroup;

    constructor(){
        var _builder= new FormBuilder();
        this.formUserGroup=_builder.group({});
        this.formUserGroup.addControl("ProductNameControl",new FormControl('',Validators.required));
        this.formUserGroup.addControl("ProductDescriptionControl",new FormControl('',Validators.required));

        this.formUserGroup.addControl("ProductPriceControl",new FormControl('',Validators.required));
        this.formUserGroup.addControl("ProductBrandControl",new FormControl('',Validators.required));
        this.formUserGroup.addControl("ProductCategoryControl",new FormControl('',[Validators.required]));
        this.formUserGroup.addControl("ProductQuantityControl",new FormControl('',Validators.required));


    }
}

File: Binder/Model/Models/Product.cs
using System;
using System.Collections.Generic;

namespace Models;

public partial class Product
{
    public int Id { get; set; }

    public string Name { get; set; }

    public string Description { get; set; }

    public int Price { get; set; }

    public int BrandId { get; set; }

    public int CategoryId { get; set; }

    public int Quantity { get; set; }

    public byte[] Image { get; set; }

    public virtual Category Category { get; set; }

    public virtual Brand IdNavigation { get; set; }

    public virtual ICollection<OrderProduct> OrderProducts { get; set; } = new List<OrderProduct>();
}
